torch/_inductor/codegen/cuda/cuda_env.py

```diff
@@ -0,0 +1,107 @@
+import functools
+from subprocess import PIPE, Popen
+
+from ... import config
+
+
+def _detect_cuda_arch_with_nvidia_smi() -> str:
+    try:
+        proc = Popen(
+            ["nvidia-smi", "--query-gpu=gpu_name", "--format=csv"],
+            stdout=PIPE,
+            stderr=PIPE,
+        )
+        stdout, stderr = proc.communicate()
+        stdout = stdout.decode("utf-8")
+        sm_names = {
+            "70": ["V100"],
+            "75": ["T4", "Quadro T2000"],
+            "80": ["PG509", "A100", "A10G", "RTX 30", "A30", "RTX 40"],
+            "90": ["H100"],
+        }
+        for sm, names in sm_names.items():
+            if any(name in stdout for name in names):
+                return sm
+        return None
+    except Exception:
+        return None
+
+
+def _detect_cuda_version_with_nvidia_smi() -> str:
+    try:
+        proc = Popen(
+            ["nvidia-smi -q | grep -i 'CUDA Version'"],
+            stdout=PIPE,
+            stderr=PIPE,
+            shell=True,
+        )
+        stdout, stderr = proc.communicate()
+        stdout = stdout.decode("utf-8")
+        version_str = stdout.split(":")[1].strip()
+        return version_str
+    except Exception:
+        return None
```

Comment:
Maybe `torch.version.cuda`?

Though do we need the host side version?

Reply:
Yeah it works! I think we just need the CUDA version.

Reply:
CUDA version the `torch` wheel was compiled with is not necessary the same as on the host? And we're going to rely on the host version to compile CUTLASS code?